Errors in tests

I am constantly running the tests of the Apps in the CI to see if I can reproduce the errors. I am compiling in FullDebug with `GLIBCXX_DEBUG` and run the small-suite with 2 omp threads
FYI @KratosMultiphysics/technical-committee 

I found one in the Fluid (@rubenzorrilla I guess this is yours), which happens randomly
~~~
testEmbeddedAusasDevelopmentCouette2D (embedded_couette_test.EmbeddedCouetteTest) ... /usr/include/c++/5/debug/safe_iterator.h:758:error: attempt to compute the 
    difference between a singular iterator to a singular iterator.

Objects involved in the operation:
iterator "lhs" @ 0x0x7f630149e7d0 {
type = N11__gnu_debug14_Safe_iteratorIN9__gnu_cxx17__normal_iteratorIPKSt4pairIPKN6Kratos12VariableDataEPvENSt9__cxx19986vectorIS9_SaIS9_EEEEENSt7__debug6vectorIS9_SE_EEEE (constant iterator);
  state = singular;
  references sequence with type `NSt7__debug6vectorISt4pairIPKN6Kratos12VariableDataEPvESaIS7_EEE' @ 0x0x7f62fc0672e8
}
iterator "rhs" @ 0x0x7f630149e7a0 {
type = N11__gnu_debug14_Safe_iteratorIN9__gnu_cxx17__normal_iteratorIPKSt4pairIPKN6Kratos12VariableDataEPvENSt9__cxx19986vectorIS9_SaIS9_EEEEENSt7__debug6vectorIS9_SE_EEEE (constant iterator);
  state = singular;
  references sequence with type `NSt7__debug6vectorISt4pairIPKN6Kratos12VariableDataEPvESaIS7_EEE' @ 0x0x7f62fc0672e8
}
~~~

I will report more if I find more:

--- a/applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp
+++ b/applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp
@@ -1350,13 +1350,12 @@ int SolidShellElementSprism3D6N::Check(const ProcessInfo& rCurrentProcessInfo) c
     KRATOS_TRY;
 
     /* Check the neighbours have been calculated */
-    // Neighbour elements
-    const GlobalPointersVector< Element >& p_neighbour_elements = this->GetValue(NEIGHBOUR_ELEMENTS);
-    KRATOS_ERROR_IF(p_neighbour_elements.size() == 0) << "The neighbour elements are not calculated" << std::endl;
-
     // Neighbour nodes
-    const WeakPointerVectorNodesType& p_neighbour_nodes = this->GetValue(NEIGHBOUR_NODES);
-    KRATOS_ERROR_IF(p_neighbour_nodes.size() == 0) << "The neighbour nodes are not calculated" << std::endl;
+    KRATOS_ERROR_IF_NOT(this->Has(NEIGHBOUR_NODES)) << "The neighbour nodes are not calculated" << std::endl;
+    if (this->Has(NEIGHBOUR_NODES)) {
+        const WeakPointerVectorNodesType& p_neighbour_nodes = this->GetValue(NEIGHBOUR_NODES);
+        KRATOS_ERROR_IF(p_neighbour_nodes.size() == 0) << "The neighbour nodes calculated are empty" << std::endl;
+    }
 
     const int check = BaseType::Check(rCurrentProcessInfo);
 

